storage/remote/client.go

```diff
@@ -209,8 +214,11 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 	if httpResp.StatusCode/100 == 5 {
 		return RecoverableError{err, defaultBackoff}
 	}
-	if httpResp.StatusCode == http.StatusTooManyRequests {
-		return RecoverableError{err, retryAfterDuration(httpResp.Header.Get("Retry-After"))}
+	if c.retryOnRateLimit {
+		if httpResp.StatusCode == http.StatusTooManyRequests {
+			return RecoverableError{err, retryAfterDuration(httpResp.Header.Get("Retry-After"))}
+		}
+		return RecoverableError{err, defaultBackoff}
```

Comment:
This would cause everything to be a recoverable error if retryOnRateLimit is on wouldn't it?

I think this whole if could just be
```
	if c.retryOnRateLimit && httpResp.StatusCode == http.StatusTooManyRequests {
	...
	}
	return err
```

Reply:
I thought the config mentions that it will activate the retry, and retry as such is active only in recoverable error. That means, if its false, it should be non-recoverable, and if true, it should be recoverable. I thought this and did accordingly. Happy to change.

Reply:
the code as is makes any http error recoverable, not only http 500 or http 429.

Reply:
Oh yes, sorry about that.

Reply:
Anyways, I have updated it.